src/deluge/gui/views/timeline_view.cpp

```diff
@@ -22,9 +22,14 @@
 #include "hid/buttons.h"
 #include "hid/display/display.h"
 #include "hid/led/pad_leds.h"
+#include "model/settings/runtime_feature_settings.h"
 #include "model/song/song.h"
 #include "processing/engines/audio_engine.h"
 #include "string.h"
+#include <algorithm>
+
+uint32_t timeHorizontalZoomLastHitClipMax;
+int8_t delayedHorizontalScrollDirection;
```

Comment:
Not sure if introducing these variables at this scope is the best way to do this (I'm not really a C/C++ dev).

Any thoughts on refactoring this for idiomaticy would be appreciated.

Reply:
they should be members of the TimelineView class probably, though that replicates it across all timeline views... as long as it doesn't end up behaving weird I think it should go in the class.